### DISCUSSION TITLE
Instant Translation function

### DISCUSSION THREAD

**Discussion by ahmedfarrag17** [espanso/espanso#1258] [Feature requests and Ideas]
I know it has little to do with the purpose of Espanso but.... why not Add the ability to translate and substitute the translated text automatically (after setting a default language in Settings for this feature), with a single shortcut. (with Google Translate, Bing or whatever).

Example:
I select "Ciao" word on any document, and once I press a Keyboard shortcut, It is automatically substituted with "Hello". (Not just relying on Google but also it can be nice to be able to add a custom library/dictionary (something similar to Matches, with current YAML approach) specifically as a translation tool, it means a simpler syntax to add new words into a specific .yml file, for ex. Hello : Ciao (it can translate in both directions, so if I write Hello it substitute it with Ciao and vice-versa), instead of 
`Trigger: Ciao
Replace: Hello` 
approach! 

It can be a nice feature! 

Honestly I have more additional ideas for building a simple and lightweight instant translate tool but If I suggest them all I know it won't make sense for the current type of usage of Espanso and why it was born in the first place. But if you'd like, Federico and Espanso lovers, I can write them down here.

Thanks a lot!

**Comment by federico-terzi** [comment]
Hey @ahmedfarrag17,

Thank you for the feedback! This use-case sounds interesting, we can definitely investigate further :)

> Honestly I have more additional ideas for building a simple and lightweight instant translate tool but If I suggest them all I know it won't make sense for the current type of usage of Espanso and why it was born in the first place. But if you'd like, Federico and Espanso lovers, I can write them down here.

Sure, please go ahead! I love to hear use-cases :)

Cheers :)
